src/app_engine/encryptor.py

```diff
@@ -0,0 +1,170 @@
+"""AES Encryption module.
+"""
+
+import logging
+import os
+import struct
+
+from Crypto import Random
+from Crypto.Cipher import AES
+from Crypto.Hash import SHA256
+
+AES_KEY = None
+AES_KEY_PATH = 'aes_key'
+HASH_SALT = None
+HASH_SALT_PATH = 'hash_salt'
+
+
+def read_file_contents(file_name):
+  """Reads contents of file at path and returns it."""
+  try:
+    path = os.path.join(os.path.dirname(__file__), file_name)
+    contents = None
+    with open(path, 'r') as f:
+      contents = f.read()
+    if not contents:
+      logging.error('Empty file at %s', path)
+      return None
+    return contents
+  except IOError as e:
+    logging.error('Failed to open file at %s', path)
+    return None
+
+
+def get_aes_key():
+  """Reads the AES key from disk."""
+  global AES_KEY
+  if AES_KEY is not None:
+    return AES_KEY
+  key = read_file_contents(AES_KEY_PATH)
+  if key is None or len(key) not in AES.key_size:
+    logging.error('Failed to load AES key.')
+  else:
+    AES_KEY = key
+  return AES_KEY
+
+
+def get_hash_salt():
+  """Reads the hash salt from disk."""
+  global HASH_SALT
+  if HASH_SALT is not None:
+    return HASH_SALT
+  salt = read_file_contents(HASH_SALT_PATH)
+  if salt is None:
+    logging.error('Failed to load hash salt.')
+  else:
+    HASH_SALT = salt
+  return HASH_SALT
+
+
+class Encryptor(object):
+  """Convenience class used to encrypt, decrypt and verify a message string.
+  """
+
+  @classmethod
+  def encrypt(cls, message):
+    """Encrypts and signs message.
+
+    Encrypted message has the following format:
+    iv + encrypt(digest + message_length + message + padding)
+    where digest is SHA256(message_length + message + padding).
+    Padding is required because AES required 16-byte aligned content.
+
+    Args:
+      message: A message string.
+
+    Returns:
+      An encrypted and signed byte string.
+    """
+    key = get_aes_key()
+    if key is None:
```

Comment:
I'm not sure what the right thing to do here is, but returning the message back in plaintext seems dangerous. Maybe we should only allow this mode with a flag? It seems safer to refuse to run in the default case if the key is missing.